File: test/aoj-dsl-2-b.test.cpp
#define PROBLEM "https://onlinejudge.u-aizu.ac.jp/problems/DSL_2_B"

#include <iostream>

#include "../cpp/segtree.hpp"

int main(void) {
    int n, q; std::cin >> n >> q;
    RSumQ<int> seg(n);
    while(q--) {
        int com, x, y; std::cin >> com >> x >> y;
        if(com == 0) seg.apply(x, y);
        else std::cout << seg.prod(x, y + 1) << '\n';
    }
}

File: test/aoj-dsl-2-b.test.py
# verification-helper: PROBLEM https://onlinejudge.u-aizu.ac.jp/problems/DSL_2_B
import sys
input = sys.stdin.buffer.readline

from py.segtree import RSumQ


def main() -> None:
    n, q = map(int, input().split())
    seg = RSumQ(n)
    for _ in range(q):
        com, x, y = map(int, input().split())
        if com == 0:
            seg.apply(x-1, y)
        else:
            print(seg.prod(x-1, y))


if __name__ == "__main__":
    main()
